src/test/scala/mesosphere/marathon/integration/setup/SingleMarathonIntegrationTest.scala

```diff
@@ -160,11 +160,42 @@ trait SingleMarathonIntegrationTest
     file.getAbsolutePath
   }
 
-  def appProxy(appId: PathId, versionId: String, instances: Int, withHealth: Boolean = true, dependencies: Set[PathId] = Set.empty): AppDefinition = {
+  def appProxy(appId: PathId, versionId: String, instances: Int, withHealth: Boolean = true, dependencies: Set[PathId] = Set.empty, docker: Boolean = false): AppDefinition = {
     val mainInvocation = appProxyMainInvocation
-    val exec = Some(s"""echo APP PROXY $$MESOS_TASK_ID RUNNING; $mainInvocation $appId $versionId http://localhost:${config.httpPort}/health$appId/$versionId""")
     val health = if (withHealth) Set(HealthCheck(gracePeriod = 20.second, interval = 1.second, maxConsecutiveFailures = 10)) else Set.empty[HealthCheck]
-    AppDefinition(appId, exec, executor = "//cmd", instances = instances, cpus = 0.5, mem = 128.0, healthChecks = health, dependencies = dependencies)
+
+    if (docker) {
```

Comment:
since we branch for most of the method and you are actually not even using the "shared" health part, I would suggest two different methods. What do you think?